raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1490,50 +1390,36 @@ private FetchRequestData buildFetchRequest() {
             .setReplicaId(quorum.localId);
     }
 
-    private void maybeSendFetch(long currentTimeMs, int leaderId) throws IOException {
+    private void maybeSendLeaderFetch(long currentTimeMs, int leaderId) throws IOException {
         maybeSendRequest(currentTimeMs, leaderId, this::buildFetchRequest);
     }
 
-    private FindQuorumRequestData buildFindQuorumRequest() {
-        return new FindQuorumRequestData()
-            .setReplicaId(quorum.localId)
-            .setHost(advertisedListener.getHostString())
-            .setPort(advertisedListener.getPort())
-            .setBootTimestamp(bootTimestamp);
-    }
+    private void maybeSendVoterFetch(long currentTimeMs) throws IOException {
+        // TODO: After a fetch timeout, do we want to avoid sending the request
+        // to the current leader?
 
-    private void maybeSendFindQuorum(long currentTimeMs) throws IOException {
-        // Find a ready member of the quorum to send FindLeader to. Any voter can receive the
-        // FindQuorum, but we only send one request at a time.
-        OptionalInt readyNodeIdOpt = connections.findReadyBootstrapServer(currentTimeMs);
-        if (readyNodeIdOpt.isPresent()) {
-            maybeSendRequest(currentTimeMs, readyNodeIdOpt.getAsInt(), this::buildFindQuorumRequest);
+        OptionalInt readyVoterIdOpt = connections.findReadyVoter(currentTimeMs);
```

Comment:
Is the purpose of this fetch for leader discovery? Not sure this is necessary as we are delaying the leader rediscovery.

Reply:
Not sure what do you mean as by it may be delaying the rediscovery?

Reply:
Not sure I follow what you mean. The intent is to only send one fetch at a time. We could send fetches to all available voters, but that makes leader changes even noisier when the number of observers is high.